styx-scheduler-service/src/test/java/com/spotify/styx/SystemTest.java

```diff
@@ -592,7 +593,7 @@ public void retriesUseLatestWorkflowSpecification() throws Exception {
     timePasses(59, SECONDS);
     timePasses(StyxScheduler.DEFAULT_SCHEDULER_TICK_INTERVAL.getSeconds() * 2, SECONDS);
 
-    awaitNumberOfDockerRuns(2);
+    await().atMost(30, SECONDS).until(() -> dockerRuns.size() >= 2);
```

Comment:
What about creating a `awaitAtMostNumberOfDockerRuns` method in the supper class to make it more in tune with the rest of the Test? Or maybe modify `awaitNumberOfDockerRuns` to accept a `IntPredicate`

```java
  void awaitNumberOfDockerRuns(IntPredicate p) {
    await().atMost(30, SECONDS).until(() -> p.test(dockerRuns.size()));
  }
```

Reply:
👍